### ISSUE TITLE
No (documented) shorthand way to copy nested objects in Object Construction

### ISSUE THREAD

**Issue by garethhumphriesgkc** [jqlang/jq#2578]
**Describe the bug**
We can easily copy fields from the root of the input object into our output by using `{ field }` syntax, but there is no way to copy nested fields with the same shorthand.  I can't say `{ fields.field }`, I must use the cumbersome `{ fields: { field: .fields.field } }`.

This is especially tedious when your desired field is many layers deep and you want to keep it in context.

While I appreciate there can be questions around what the expected output is when navigating nested objects, I think in this case the same short hand logic makes unambiguous sense - it is still "create a field called _this_, with the contents of _this_", it's just that now _this_ has multiple layers to it.

**To Reproduce**
```
$ echo '{"a":{"b":1}}' | jq '{ a }'
{
  "a": {
    "b": 1
  }
}
$ echo '{"a":{"b":1}}' | jq '{ a.b }'
jq: error: syntax error, unexpected FIELD, expecting '}' (Unix shell quoting issues?) at <top-level>, line 1:
{a.b}  
jq: 1 compile error
$
```

**Expected behavior**
Output should include the value of the selected field
```
$ echo '{"a":{"b":1,"junk":0},"useless":0}' | jq '{ a.b }'
{
  "a": {
    "b": 1
  }
}
$
```

**Environment (please complete the following information):**
jq-1.6

**Additional context**

**Comment by wader** [comment]
Hi, yeap something like that would be nice. jq development and maintenance is current a bit on hold, see s#2305 

The closest i think of with current version of jq is a `pick` function described here https://github.com/stedolan/jq/issues/2062#issuecomment-1165579448

**Comment by pkoppstein** [comment]
Here's a version of `pick` that works with both jq and gojq, and allows multiple dot-paths to be specified, but does NOT check whether the incoming entity has the specified paths:
```
# stream should be a stream of dot-paths
def pick(stream):
  . as $in
  | reduce path(stream) as $a (null;
      setpath($a; $in|getpath($a)) );
```
Example:
```
{a: {b: 1, c: 2, d: {e: 2}}} | pick(.a.b, .a.d.e)
```
yields
```
{"a":{"b":1,"d":{"e":2}}}
```

**Comment by itchyny** [comment]
`pick/1` is implemented by #2656 and merged so closing.
